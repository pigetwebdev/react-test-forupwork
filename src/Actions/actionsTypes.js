export const MOVE_NEXT_PAGE = 'MOVE_NEXT_PAGE'
export const MOVE_LAST_PAGE = 'MOVE_LAST_PAGE'