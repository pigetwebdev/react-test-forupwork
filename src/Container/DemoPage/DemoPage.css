.DemoPageWrapper{
    padding: 50px;
    width: 100%;
    height: 100%;
    display: block;
    background-image: url('../../asset/img_back.png');
    background-size: cover;
 
}