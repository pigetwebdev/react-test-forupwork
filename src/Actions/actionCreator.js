import { MOVE_NEXT_PAGE } from './actionsTypes'
import { MOVE_LAST_PAGE } from './actionsTypes'

let CurPageIndex = 0; //First Page
export const moveNextPage = () => ({
    type: MOVE_NEXT_PAGE,
    PageIndex: CurPageIndex < 2? ++CurPageIndex: 2
})

export const moveLastPage = () => ({
    type: MOVE_LAST_PAGE,
    PageIndex: 2
})
