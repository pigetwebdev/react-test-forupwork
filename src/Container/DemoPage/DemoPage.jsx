import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
//import { Test } from './DemoPage.styles';
import DemoChatBox from '../../Component/DemoChatBox'
import DemoLikeProduct from '../../Component/DemoLikeProduct'
import DemoLastPage from '../../Component/DemoLastPage'
import './DemoPage.css';
class DemoPage extends PureComponent { 
  constructor(props) {
    super(props);

    this.state = {
      demoState: 0,
    };
  }

  render () {
    return (
      <div className="DemoPageWrapper">
				{/* <img src={backImg} className="back-img" alt="" /> */}
				{(this.props.pageIndex === 0)&&<DemoChatBox />}
				{(this.props.pageIndex === 1)&&<DemoLikeProduct />}
				{(this.props.pageIndex === 2)&&<DemoLastPage />}
      </div>
    );
  }
}

const mapStateToProps = state => {
  return { pageIndex: state.movePageIndex
 };
};
export default connect(
  mapStateToProps,
  null,
)(DemoPage);
