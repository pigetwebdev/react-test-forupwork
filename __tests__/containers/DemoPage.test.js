import React from 'react';
import DemoPage from '../../src/Container/DemoPage';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import { moveNextPage, moveLastPage } from '../../src/Actions/actionCreator';
import configureStore from 'redux-mock-store'
const storeFake = state => {
  return {
    default: jest.fn(),
    subscribe: jest.fn(),
    dispatch: jest.fn(),
    getState: () => state,
  };
};

describe('>>>Demopage --- Snapshot',()=>{
  it('+++capturing Snapshot of DemoPage', () => {
    const store = storeFake({});  
      const renderedValue =  renderer.create(
        <Provider store={store}>
        <DemoPage />
        </Provider>).toJSON()
      expect(renderedValue).toMatchSnapshot();
  });

});
