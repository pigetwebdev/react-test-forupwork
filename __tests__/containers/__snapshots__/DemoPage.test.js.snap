// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Demopage --- Snapshot +++capturing Snapshot of DemoPage 1`] = `
<div
  className="DemoPageWrapper"
/>
`;
